msal/src/Microsoft.Identity.Client/Internal/Requests/InteractiveRequest.cs

```diff
@@ -285,6 +285,11 @@ private void CheckForDuplicateQueryParameters(string queryParams, IDictionary<st
                 authorizationRequestParameters[kvp.Key] = kvp.Value;
             }
 
+            if (_uiBehavior.PromptValue == UIBehavior.NoPrompt.PromptValue)
+            {
+                return authorizationRequestParameters;
+            }
+
             authorizationRequestParameters[OAuth2Parameter.Prompt] = _uiBehavior.PromptValue;
```

Comment:
this is a bit of a special scenario.... I think to best ensure that this doesn't break some time in the future (by someone not returning anymore....) 
I suggest that you invert the assignment and then only assign the prompt in case the uibehavior is not NoPrompt. makes sense?